code/__HELPERS/global_lists.dm

```diff
@@ -97,9 +97,13 @@ var/global/list/datum/poster/poster_designs = subtypesof(/datum/poster)
 	for(var/X in subtypesof(/datum/xeno_caste))
 		var/datum/xeno_caste/C = new X
 		if(!(C.caste_type_path in GLOB.xeno_caste_datums))
-			GLOB.xeno_caste_datums[C.caste_type_path] = list(1,2,3,4)
-		var/upgrade_level = CLAMP(C.upgrade + 1, 1, 4)
-		GLOB.xeno_caste_datums[C.caste_type_path][upgrade_level] = C
+			GLOB.xeno_caste_datums[C.caste_type_path] = list()
+		GLOB.xeno_caste_datums[C.caste_type_path][C.upgrade] = C
+
+	for(var/H in subtypesof(/datum/hive_status))
+		var/datum/hive_status/HS = new H
+		GLOB.hive_datums += HS.hivenumber
+		GLOB.hive_datums[HS.hivenumber] = HS
```

Comment:
I hope `subtypesof(/datum/hive_status)` are ordered as in 1, 2, 3... or this will runtime.

Reply:
They are.

Reply:
Should I change hivenumber to a string?

Reply:
The problem is that it's hiding the fact that the list is accessed by index (hence the needed `GLOB.hive_datums += HS.hivenumber` first... making `GLOB.hive_datums[HS.hivenumber] = HS` useless).

Since everything else is accessed by associative value, it may indeed be less error-prone to switch `hivenumber` to a string and access by value, except if `hivenumber` is used a number somewhere in the code.

Reply:
It isn't, i'll change it to a string